{"numFailedTestSuites":1,"numFailedTests":1,"numPassedTestSuites":0,"numPassedTests":18,"numPendingTestSuites":0,"numPendingTests":0,"numRuntimeErrorTestSuites":0,"numTodoTests":0,"numTotalTestSuites":1,"numTotalTests":19,"openHandles":[],"snapshot":{"added":0,"didUpdate":false,"failure":false,"filesAdded":0,"filesRemoved":0,"filesRemovedList":[],"filesUnmatched":0,"filesUpdated":0,"matched":0,"total":0,"unchecked":0,"uncheckedKeysByFile":[],"unmatched":0,"updated":0},"startTime":1758350390060,"success":false,"testResults":[{"assertionResults":[{"ancestorTitles":["Random key generation"],"duration":1,"failureDetails":[],"failureMessages":[],"fullName":"Random key generation generates keys at random","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"generates keys at random"},{"ancestorTitles":["Random key cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Random key cipher has a key made of letters","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"has a key made of letters"},{"ancestorTitles":["Random key cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Random key cipher has a key that is at least 100 characters long","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"has a key that is at least 100 characters long"},{"ancestorTitles":["Random key cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Random key cipher can encode","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"can encode"},{"ancestorTitles":["Random key cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Random key cipher can decode","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"can decode"},{"ancestorTitles":["Random key cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Random key cipher is reversible","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"is reversible"},{"ancestorTitles":["Incorrect key cipher"],"duration":5,"failureDetails":[],"failureMessages":[],"fullName":"Incorrect key cipher throws an error with an all caps key","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"throws an error with an all caps key"},{"ancestorTitles":["Incorrect key cipher"],"duration":1,"failureDetails":[],"failureMessages":[],"fullName":"Incorrect key cipher throws an error with a mixed case key","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"throws an error with a mixed case key"},{"ancestorTitles":["Incorrect key cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Incorrect key cipher throws an error with a numeric key","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"throws an error with a numeric key"},{"ancestorTitles":["Incorrect key cipher"],"duration":1,"failureDetails":[{"matcherResult":{"message":"expect(received).toThrow(expected)\n\nExpected substring: \"Bad key\"\n\nReceived function did not throw","pass":false}}],"failureMessages":["Error: expect(received).toThrow(expected)\n\nExpected substring: \"Bad key\"\n\nReceived function did not throw\n    at Object.<anonymous> (/app/simple-cipher.test.ts:69:8)\n    at Promise.then.completed (/base/node_modules/jest-circus/build/utils.js:298:28)\n    at new Promise (<anonymous>)\n    at callAsyncCircusFn (/base/node_modules/jest-circus/build/utils.js:231:10)\n    at _callCircusTest (/base/node_modules/jest-circus/build/run.js:316:40)\n    at _runTest (/base/node_modules/jest-circus/build/run.js:252:3)\n    at _runTestsForDescribeBlock (/base/node_modules/jest-circus/build/run.js:126:9)\n    at _runTestsForDescribeBlock (/base/node_modules/jest-circus/build/run.js:121:9)\n    at run (/base/node_modules/jest-circus/build/run.js:71:3)\n    at runAndTransformResultsToJestFormat (/base/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)\n    at jestAdapter (/base/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)\n    at runTestInternal (/base/node_modules/jest-runner/build/runTest.js:367:16)\n    at runTest (/base/node_modules/jest-runner/build/runTest.js:444:34)"],"fullName":"Incorrect key cipher throws an error with an empty key","invocations":1,"location":null,"numPassingAsserts":0,"retryReasons":[],"status":"failed","title":"throws an error with an empty key"},{"ancestorTitles":["Substitution cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Substitution cipher keeps the submitted key","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"keeps the submitted key"},{"ancestorTitles":["Substitution cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Substitution cipher can encode","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"can encode"},{"ancestorTitles":["Substitution cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Substitution cipher can decode","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"can decode"},{"ancestorTitles":["Substitution cipher"],"duration":1,"failureDetails":[],"failureMessages":[],"fullName":"Substitution cipher is reversible","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"is reversible"},{"ancestorTitles":["Substitution cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Substitution cipher : double shift encode","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":": double shift encode"},{"ancestorTitles":["Substitution cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Substitution cipher can wrap on encode","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"can wrap on encode"},{"ancestorTitles":["Substitution cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Substitution cipher can wrap on decode","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"can wrap on decode"},{"ancestorTitles":["Substitution cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Substitution cipher can handle messages longer than the key (encode)","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"can handle messages longer than the key (encode)"},{"ancestorTitles":["Substitution cipher"],"duration":0,"failureDetails":[],"failureMessages":[],"fullName":"Substitution cipher can handle messages longer than the key (decode)","invocations":1,"location":null,"numPassingAsserts":1,"retryReasons":[],"status":"passed","title":"can handle messages longer than the key (decode)"}],"endTime":1758350391371,"message":"  ● Incorrect key cipher › throws an error with an empty key\n\n    expect(received).toThrow(expected)\n\n    Expected substring: \"Bad key\"\n\n    Received function did not throw\n\n      67 |     expect(() => {\n      68 |       new SimpleCipher('');\n    > 69 |     }).toThrow('Bad key');\n         |        ^\n      70 |   });\n      71 | });\n      72 |\n\n      at Object.<anonymous> (simple-cipher.test.ts:69:8)\n","name":"/app/simple-cipher.test.ts","startTime":1758350390165,"status":"failed","summary":""}],"wasInterrupted":false}
